software/contrib/diagnostic.py

```diff
@@ -69,9 +86,8 @@ def main(self):
             oled.fill(0)
 
             # calc and format temp
-            use_fahrenheit = b1.value() or b2.value()
             t = self.calc_temp()
-            formatted_temp = f"{int(self.convert_fahrenheit(t) if use_fahrenheit else t)}{'F' if use_fahrenheit else 'C'}"
+            formatted_temp = f"{int(t)}{self.config['temp_units']}"
```

Comment:
[discussion] In the constructor you have the line:
```python
self.use_fahrenheit = self.config["temp_units"] == "F"
```
I think I like the idea of setting class instance value from the config in the constructor instead of referencing `self.config` throughout the script. What if we set that precident here?

```python
def __init__(self):
    ....
    self.temp_units = self.config['temp_units']
```
Then we use that instance variable in the script:
```suggestion
            formatted_temp = f"{int(t)}{self.temp_units}"
```

EDIT: after reading down further, I see that's how you've done that with the Turing Machine script, awesome! I still think it'd be a good idea to do that here too.